infra/RolesStack.py

```diff
@@ -0,0 +1,84 @@
+from aws_cdk import (
+    aws_iam as iam,
+    CfnOutput,
+    Stack,
+)
+import boto3
+from constructs import Construct
+
+DATA_PIPELINE_LAMBDA_EXECUTION_ROLE_PATTERN = "maap-data-pipelines-*-datapipelinelambdarole*"
+STAC_INGESTOR_EXECUTION_ROLE_PATTERN = 'MAAP-STAC-*-pgSTAC-stacingestorexecutionrole*'
+
+class RolesStack(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+        
+        self.create_data_access_role()
+
+        # export a cloud formation output with the data access role name and arn
+        CfnOutput(
+            self,
+            "data access role arn",
+            export_name=f"data-access-role-arn",
+            value=self.data_access_role.role_arn
+        )
+            
+    def create_data_access_role(
+        self
+    ):
+        """
+        Creates data access role, attaches inline policy to allow access to s3 buckets, and grants assume role to data pipeline and stac ingestor roles
+        """
+        
+        role = iam.Role(
+            self,
+            "data-access-role",
+            assumed_by=iam.ServicePrincipal("lambda.amazonaws.com")
+        )
+        
+        
+        buckets = [
+                "nasa-maap-data-store",
+                "maap-user-shared-data",
+                "maap-ops-workspace",
+                "maap-data-store-test",
+                "nasa-maap-data-store/*",
+                "maap-user-shared-data/*",
+                "maap-ops-workspace/*",
+                "maap-data-store-test/*"
+            ]
+        
+        role.attach_inline_policy(
+            iam.Policy(
+                self,
+                "Policy",
+                statements=[
+                    iam.PolicyStatement(
+                        effect=iam.Effect.ALLOW,
+                        actions=["s3:PutObject*", "s3:ListBucket*", "s3:GetObject*"],
```

Comment:
Is PutObjects needed?

Reply:
> Is PutObjects needed for this specific role

I copy pasted these actions from the hard coded policy in the UAH account, without thinking much about what's actually needed. AFAIK this role is used in : 

- stac-ingestor 
- build-stac
- might in the future be used in the tiler deployments. 

None of these need write permissions. If we don't see any reason for this permission to be required in the future I can remove it.